testing/asserts.ts

```diff
@@ -749,9 +749,17 @@ export async function assertRejects<E extends Error = Error>(
     msg = msgIncludesOrMsg;
   }
   let doesThrow = false;
+  let isPromiseReturned = false;
   try {
-    await fn();
+    const possiblePromise = fn();
+    if (possiblePromise instanceof Promise) {
+      isPromiseReturned = true;
+      await possiblePromise;
+    }
   } catch (error) {
+    if (!isPromiseReturned) {
+      throw new AssertionError("An error was thrown but a promise was not rejected.");
```

Comment:
For `assertError`, we return an error like this when an error is not thrown.
```ts
if (!doesThrow) {
  msg = `Expected function to throw${msg ? `: ${msg}` : "."}`;
  throw new AssertionError(msg);
}
```
I think we should have a similar message for when it was expected to reject. Maybe "Function throws when expected to reject".

Also the messages later in this assertRejects function should be updated.

- "A non-Error object was thrown or rejected." should be "A non-Error object was rejected."
- "Expected function to throw" should be "Expected function to reject"